drivers/gpio/Kconfig

```diff
@@ -15,6 +15,11 @@ module = GPIO
 module-str = gpio
 source "subsys/logging/Kconfig.template.log_config"
 
+config GPIO_GET_DIRECTION_BITS
+	bool
```

Comment:
Why do we need this Kconfig?

Reply:
To indicate whether a driver supports getting pin direction. Support for this feature was requested to be optional in a previous iteration of this change from a long while ago

Reply:
If the API returns `ENOTSUP`, wouldn't that indicate whether this is supported or not supported?

Reply:
I would prefer to remove it as well, personally, but I think with the GPIO API, it would require a fair amount of work to implement the feature for each platform. In a lot of applications, this capability isn't necessary either, so making it configurable would reduce some amount of code footprint.

Reply:
@mmahadevan108 - you do bring up a good point though. One thing I had not really considered was the case where are are multiple GPIO drivers - e.g. the SoC GPIO driver and then something like a GPIO expander. Globally, the API might be enabled, but still not all platforms would be able to support it.